src/com/hms_networks/americas/sc/taginfo/TagInfoManager.java

```diff
@@ -282,22 +301,194 @@ private static synchronized void processTagListEBDLine(String line) {
+              tagInGroupD,
+              tagTypeObj);
+    }
+    tagInfoList[tagInfoListInsertIndex] = currentTagInfo;
+    tagInfoListInsertIndex++;
+  }
+
+  /**
+   * Reads the enumeration file to get a list of tags that have int to string enumerations and
+   * stores the list. This must be done once at the start of the application and is handled when the
+   * TagInfoList is created.
+   *
+   * @throws IOException if int to string enumeration file read fails
+   * @throws JSONException if int to string enumeration JSON parse fails
+   */
+  private static void getIntStringEnumTags() throws IOException, JSONException {
+    if (enableIntToStringEnums) {
+      ArrayList tagList = new ArrayList();
+      JSONObject allTagIntToStringMappingsJson;
+      JSONArray tagIntToStringMappingsJsonArray;
+      allTagIntToStringMappingsJson =
+          new JSONObject(FileAccessManager.readFileToString(INT_TO_STRING_ENUMERATION_FILE_NAME));
+      tagIntToStringMappingsJsonArray =
+          (JSONArray) allTagIntToStringMappingsJson.get("enumeratedTagList");
+      final int tagListJsonArrayLength = tagIntToStringMappingsJsonArray.length();
+      for (int i = 0; i < tagListJsonArrayLength; i++) {
+        String tagValueAsString = tagIntToStringMappingsJsonArray.getString(i);
+        tagList.add(tagValueAsString);
+      }
+      intToStringEnumerationTagList = tagList;
+    }
+  }
+
+  /**
+   * For a single tag, get the array of string mappings corresponding to the given integer tag.
+   *
+   * @param tagName the name of the Ewon tag
+   * @return an array of string mappings corresponding to the current integer tag
+   * @throws IOException if int to string enumeration file read fails
+   * @throws JSONException if int to string enumeration JSON parse fails
+   */
+  private static String[] getIntStringEnumTagMappings(String tagName)
+      throws IOException, JSONException {
+    String[] tagStringMappings = null;
+    JSONObject tagIntToStringMappingsFileJson;
+    JSONObject tagIntToStringMappingsJson;
+    JSONArray tagIntToStringMappingsJsonArray;
+    if (enableIntToStringEnums) {
+      tagIntToStringMappingsFileJson =
+          new JSONObject(FileAccessManager.readFileToString(INT_TO_STRING_ENUMERATION_FILE_NAME));
+      tagIntToStringMappingsJson = (JSONObject) tagIntToStringMappingsFileJson.get("tags");
+      tagIntToStringMappingsJsonArray = (JSONArray) tagIntToStringMappingsJson.get(tagName);
+      final int length = tagIntToStringMappingsJsonArray.length();
+      int biggestIndex = 0;
+
+      /* Int to string enumerations are held in a JSON array of one key/value pair for each index in the array.
+       * To get the key of a key value pair, get the first(only) item returned by JSONObject.names()
+       * for the JSON object index.
+       */
+      final int tagIntToStringEnumStringKeyIndex = 0;
+      for (int jsonArrayIndex = 0; jsonArrayIndex < length; jsonArrayIndex++) {
+        String currentKey =
```

Comment:
why store this if its not really going to be used, seems like this should just be 
```
int currentKey = Integer.parseInt(....)
```

If you want to keep it seperate just to keep the line length down there should be better naming scheme than currentKey and current (which are the same information, just different types.)

Reply:
changed the name and changed to not store the string